packages/cli/src/project-references.test.ts

```diff
@@ -184,61 +193,228 @@ describe('createProjectReferencesCompilerHost', () => {
+      const host = createProjectReferencesCompilerHost(
+        ['module'],
+        tsConfig.options,
+        getDefinedArray(program.getResolvedProjectReferences()),
+        sys,
+      );
+
+      const sourceFile = host.getSourceFile(
+        getFixture(FIXTURE_NAME, 'packages/project-1/dist/index.d.ts'),
+        ScriptTarget.ES2020,
+      );
+
+      expect(sourceFile).toBeDefined();
+      expect(sourceFile?.fileName).toContain(
+        'packages/project-1/dist/index.d.mts',
+      );
+    });
 
-    expect(sourceFile).toBeDefined();
-    expect(sourceFile?.fileName).toContain(
-      'packages/project-1/dist/index.d.cts',
-    );
+    it('does not modify the source file for non-output files', () => {
+      const host = createProjectReferencesCompilerHost(
+        ['commonjs'],
+        tsConfig.options,
+        getDefinedArray(program.getResolvedProjectReferences()),
+        sys,
+      );
+
+      const sourceFile = host.getSourceFile(
+        fileURLToPath(import.meta.url),
+        ScriptTarget.ES2020,
+      );
+
+      expect(sourceFile).toBeDefined();
+      expect(sourceFile?.fileName).toBe(fileURLToPath(import.meta.url));
+    });
   });
 
-  it('modifies the source file to `.mts` when building `module`', () => {
-    const host = createProjectReferencesCompilerHost(
-      ['module'],
-      tsConfig.options,
-      getDefinedArray(program.getResolvedProjectReferences()),
-    );
-
-    const sourceFile = host.getSourceFile(
-      getFixture('project-references', 'packages/project-1/dist/index.d.ts'),
-      ScriptTarget.ES2020,
-    );
+  describe('resolveModuleNameLiterals', () => {
+    it('resolves a `.cjs` file', () => {
+      const host = createProjectReferencesCompilerHost(
+        ['commonjs'],
+        tsConfig.options,
+        getDefinedArray(program.getResolvedProjectReferences()),
+        sys,
+      );
+
+      const containingPath = getFixture(
+        FIXTURE_NAME,
+        'packages/project-3/src/index.ts',
+      );
+
+      const containingFile = host.getSourceFile(
+        containingPath,
+        ScriptTarget.ES2020,
+      );
+
+      assert(containingFile);
+
+      const modules = host.resolveModuleNameLiterals?.(
+        [
+          factory.createStringLiteral('./index.cjs'),
+          factory.createStringLiteral('./foo.cjs'),
+        ],
+        containingPath,
+        undefined,
+        tsConfig.options,
+        containingFile,
+        [],
+      );
+
+      expect(modules).toHaveLength(2);
+      expect(modules?.[0]?.resolvedModule?.resolvedFileName).toBe(
+        getFixture(FIXTURE_NAME, 'packages/project-3/src/index.ts'),
+      );
+      expect(modules?.[1]?.resolvedModule?.resolvedFileName).toBe(
+        getFixture(FIXTURE_NAME, 'packages/project-3/src/foo.ts'),
+      );
+    });
 
-    expect(sourceFile).toBeDefined();
-    expect(sourceFile?.fileName).toContain(
-      'packages/project-1/dist/index.d.mts',
-    );
+    it('resolves a `.mjs` file', () => {
```

Comment:
I actually thought about this, but it depends on how you say ".mjs". If you say the dot too for example, it would be "a". For that reason I kept it simple. 😅

Reply:
Ah, it is indeed correct if you pronounce it "dot em jay ess". Hrm, hrm, hrm. Very well.